components/style/properties/properties.mako.rs

```diff
@@ -1091,6 +1098,10 @@ impl PropertyDeclaration {
                     match input.try(|i| CSSWideKeyword::parse(context, i)) {
                         Ok(CSSWideKeyword::InheritKeyword) => {
                             % for sub_property in shorthand.sub_properties:
+                                if properties_seen.get_${property.ident}() {
```

Comment:
The idea was to keep the `properties_seen` code next to the `result_list.push` occurences, since they kinda sorta operate on the same data. This causes some code repetition, but there already is some repeated code dealing with pushing `WideKeyword`s to the `result_list` (the 3 branches are almost identical).
As for the non-wide keywords, I missed that one - that's the only branch where pushing to the `result_list` is not explicit. I'll fix that.